leap_ec/probe.py

```diff
@@ -206,6 +208,22 @@ def __call__(self, next_individual):
             yield ind
 
 
+def _to_quoted_list(arr):
+    return f"{json.dumps(arr.tolist())}"
+
+@contextlib.contextmanager
```

Comment:
I'm curious as to why you are defining a context manager?  I see the with clauses that use them, but it's not clear to me why one is necessary?  Not saying this is a bad approach, but rather if justified that a comment explaining "why" a context manager is being used would be helpful.